enumfields/enums.py

```diff
@@ -39,6 +41,33 @@ def __new__(mcs, name, bases, attrs):
 
 class ChoiceEnum(ChoiceEnumMeta('ChoiceEnum', (BaseEnum,), _EnumDict())):
 
+    @classmethod
+    def get_allowed_previous_choices(cls, current_choice):
+        return cls._get_previous_map().get(current_choice) or []
+
+    @classmethod
+    @lru_cache(maxsize=1)
```

Comment:
why lru cache? why not cachedproperty?

Reply:
it was because I needed to cache classmethod which didn't worked with cached_property. Whole method is gone.